Fixing the Android Issue (ONCE AND FOR ALLLL)

agarg09 opened the pull request:
**Summary Of Changes**
-This took a lot of restructuring, but I’ll try to break it down as succinctly as possible:

_app.json/app.tsconfig changes:_
-changed the project id & slug so they now align with the Expo Project “Butterwalk”
-this was done so that we could resolve the issue of Android sign in not working on Expo Development Preview Builds (they were working on Expo Development - Simulator Builds)
-outside the code changes: I created a Expo project named “Butterwalk” + changed the SHA fingerprint in the Google Cloud Console Android Client (this client is called Android Client (test Adwita) for now) to now match with the new Expo’s project SHA fingerprint (new Expo project, new fingerprint)

I basically broke signin.tsx into 3 parts: since I observed/from previous weeks that Android by default tries to find a oauthredirect.tsx file, we originally tried to solve this by making this a loading page, but the problem then became that the data wasn’t being passed correctly…now with this three tier structure, data should be passed correctly and updated as well.

_signin.tsx:_
Basically only contains the SignIn page UI and the act of “signing in”, the useEffect() sends the response (in JSON format) from Google (which we need so we can parse data, such as netid) over when redirecting to oauthRedirect (I kind of just made it more explicit that hey, go to this oAuthRedirect page that Android keeps looking for)

_oAuthRedirect.tsx:_
Honestly is a bridge to the finishSignIn.tsx, it just continues to pass the jsonified response from Google (above)

_finishSignIn.tsx:_
Where all the magic happens (literally): make all the web socket connections, passing data, getting data, and reroutes to home or finishAcc page as necessary

^please follow comments in the code for more explanation, but this is kind of it, for more details, ask @Adwita!

agarg09 commented:
**Changes that still need to be done after the okay from Dev Leads**

-Delete old Android Client, the one linked to the Expo Project “husky-betterWalk” - I didn’t do this because other Devs might be using Android, so I didn’t want to delete it without permission just yet, @Snigdha & @Katharine, need the green light for this

-Just to confirm with all the Client Id info/issues and stuff, lets confirm all Client IDs match what Expo/.env files look like, and update everyone on Dev about this too, as some might be using old client IDs (also, I noticed there are variables in Expo, should we be worried about changing values there too?)

-I saw that Maps API Key for Android has a restriction on a specific SHA fingerprint, since we are now moving projects/changing SHA fingerprints, again after the green light from @Snigdha & @Katharine, lets change that SHA fingerprint restriction in Google Cloud Console

-I removed the error message thing (the red text that would pop up) (mainly because of how it got structured, like the 3-part thing, the error message is now in the finishSignIn.tsx), I can put it back in, but I thought that wasn’t the highest priority for now, but @Snigdha and @Katharine let me know I’ll have to add it a bit differently than before, but I can still add it back

-Test Users: do we want to add more Devs/“students” for Tuesday, lets do that so people aren’t blocked when doing sign in (just to be on the safe side)

-Once we have done ALL of this, I will go through and run Android and IOS builds again, @Snigdha, I can’t create a IOS preview, so I would like it if you could create it and make sure it works smoothly

agarg09 commented:
**^^ DRAFT PR ^^**

SnigdhaMah commented:
> **Changes that still need to be done after the okay from Dev Leads**
> 
> -Delete old Android Client, the one linked to the Expo Project “husky-betterWalk” - I didn’t do this because other Devs might be using Android, so I didn’t want to delete it without permission just yet, @SnigdhaMah & @KatharineZhang, need the green light for this
> 
> -Just to confirm with all the Client Id info/issues and stuff, lets confirm all Client IDs match what Expo/.env files look like, and update everyone on Dev about this too, as some might be using old client IDs (also, I noticed there are variables in Expo, should we be worried about changing values there too?)
> 
> -I saw that Maps API Key for Android has a restriction on a specific SHA fingerprint, since we are now moving projects/changing SHA fingerprints, again after the green light from @SnigdhaMah & @KatharineZhang, lets change that SHA fingerprint restriction in Google Cloud Console
> 
> -I removed the error message thing (the red text that would pop up) (mainly because of how it got structured, like the 3-part thing, the error message is now in the finishSignIn.tsx), I can put it back in, but I thought that wasn’t the highest priority for now, but @SnigdhaMah and @KatharineZhang let me know I’ll have to add it a bit differently than before, but I can still add it back
> 
> -Test Users: do we want to add more Devs/“students” for Tuesday, lets do that so people aren’t blocked when doing sign in (just to be on the safe side)
> 
> -Once we have done ALL of this, I will go through and run Android and IOS builds again, @SnigdhaMah, I can’t create a IOS preview, so I would like it if you could create it and make sure it works smoothly

Approved.

agarg09 commented:
Out of code changes:
-Deleted old Android client ID
-Confirmed all .env/client IDs match
-Changed the SHA fingerprint in the Android Google Maps API key
-Put the error message thing back (look above)^

SnigdhaMah reviewed (approved):
Finally looks good on both iOS and Android preview builds! I have run out of words in the dictionary to congratulate you but truly amazing job! I am impressed with how you were able to wade through so many issues over the three months! You should feel super proud of yourself.
(Make sure to change the name HuskyButterProject to SafeTrip on google sign in before you squash and merge!)
<img width="300" height="700" alt="IMG_1791" src="https://github.com/user-attachments/assets/8d9ac902-26f4-420f-a335-d74ac6112481" />
